employee-rostering-backend/src/main/java/org/optaweb/employeerostering/controller/SkillController.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaweb.employeerostering.controller;
+
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class SkillController {
```

Comment:
It looks like you are doing controller correctly; however, the paths do no match the old paths, which is a problem for backward compatibility. The paths should be:

```
@GetMapping(/tenantId/{tenantId}/skill/{skillId})
public String getSkill(@PathVariable int tenantId, @PathVariable int skillId)
```

```
@DeleteMapping(/tenantId/{tenantId}/skill/{skillId})
public String removeSkill(@PathVariable int tenantId, @PathVariable int skillId)
```


```
@PostMapping(/tenantId/{tenantId}/skill/update)
public String updateSkill(@PathVariable int tenantId)
```

```
@PostMapping(/tenantId/{tenantId}/skill/add)
public String addSkill(@PathVariable int tenantId)
```

You are also missing the list getter method (the one that return all the skills for a given Tenant):

```
@GetMapping(/tenantId/{tenantId}/skill)
public String getSkillList(@PathVariable int tenantId)
```

Reply:
Good catch, though note that path names must be enclosed by parentheses. I'll implement the controller methods once I have Skill database persistence going.